### PR TITLE
Feature/test Rabi cal

### PR DESCRIPTION
**Pull request by dieris**
Some initial tests for `RabiAmpCalibration`. If you are happy with this template, we'll keep adding more tests.

### CODE REVIEW

**Comment by dieris** [comment]
Added Ramsey

**Comment by grahamrow** [comment]
So the general gist is to provide `ideal_fake_data` as part of the configuration for an X6. I think I'm okay with this approach.

File: src/auspex/instruments/X6.py [hunk 1]
```diff
@@ -50,22 +50,30 @@ def __init__(self, settings_dict=None):
...
+                        self.kernel = eval(value)
+                    except:
+                        raise ValueError('Kernel invalid. Provide a file name or an expression to evaluate')
             elif name == "kernel_bias" and isinstance(value, str) and value:
                 self.kernel_bias = eval(value)
             #elif hasattr(self, name):
             #        setattr(self, name, value)
             elif name == "channel":
                 setattr(self, 'phys_channel', int(value))
+            elif name == 'ideal_fake_data': # for testing purposes
```
Comment (grahamrow) [inline src/auspex/instruments/X6.py]: Looks like the configurations below use "ideal_data" not "ideal_fake_data"?

File: src/auspex/instruments/X6.py [hunk 2]
```diff
@@ -104,6 +112,7 @@ def __init__(self, resource_name=None, name="Unlabeled X6", gen_fake_data=False)
 
         self.last_timestamp = datetime.datetime.now()
         self.gen_fake_data = gen_fake_data
+        self.ideal_data = 0
```
Comment (grahamrow) [inline src/auspex/instruments/X6.py]: Couldn't this be None?

File: src/auspex/instruments/X6.py [hunk 3]
```diff
@@ -143,6 +152,12 @@ def disconnect(self):
     def set_all(self, settings_dict):
         # Call the non-channel commands
         super(X6, self).set_all(settings_dict)
+        # Set data for testing
+        if self.ideal_data:
+            try:
+                self.ideal_data = np.load(os.path.abspath(self.ideal_data+'.npy'))
+            except:
```
Comment (grahamrow) [inline src/auspex/instruments/X6.py]: If None above, we could ditch this None here and instead throw a proper error if the file doesn't exist.

File: src/auspex/pulse_calibration.py [hunk 4]
```diff
@@ -97,7 +97,22 @@ def set(self, instrs_to_set = [], first_step = True, **params):
             extra_plot_server = self.exp.extra_plot_server
         self.exp = QubitExpFactory.create(meta_file=meta_file, calibration=True, save_data=False, cw_mode=self.cw_mode)
         self.exp.leave_plot_server_open = True
-        self.exp.first_exp = first_step
+        self.exp.first_exp = not bool(exp_step)
+
```
Comment (grahamrow) [inline src/auspex/pulse_calibration.py]: Hopefully nobody decides to use similarly named property inadvertantly 😉

File: src/auspex/instruments/X6.py (hunk 1 above)
Comment (dieris) [reply src/auspex/instruments/X6.py]: Yes. In reality, I'm only using the X6 property for now, not the channel. But may be useful later for two-qubit gates...

**Comment by dieris** [comment]
I'll add some docstrings, thanks for taking a look

**Pull request merged**